.github/workflows/build_and_deploy_dev.yml

```diff
@@ -0,0 +1,86 @@
+name: Build and Deploy to dev service app
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [elr/gh-action-dev-deploy]
+    # branches: [master]
+
+env:
+  CODE_ENVIRONMENT: dev
+  APPLICATION_TYPE: api
+  AZURE_WEBAPP_NAME: clearlydefined-api-dev
+  DOCKER_IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/${{ github.repository }}-dev
+  # SEMVER: '' # if you want to use semver, you will need to add a tag to the Docker build
+
+jobs:
+  build-and-deploy:
+    name: Build and Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Log into ghcr registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }} # user that kicked off the action
+          password: ${{ secrets.GITHUB_TOKEN }} # token created when the action launched (short lived)
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          file: Dockerfile
+          tags: |
+            ${{ env.DOCKER_IMAGE_NAME }}:latest 
```

Comment:
I'm a bit on the fence regarding whether we need 'latest' tag, not sure what good use it brings. At least for deployment I think it is an anti-pattern, see my further comment on this below.